tasks/unit-tests/components_tests.py

```diff
@@ -8,13 +8,8 @@
 
 class TestComponents(unittest.TestCase):
     def setUp(self):
-        # Add the classic component (used by tests) to the allowlist
-        classicComp = 'comp/classic/classicimpl'
-        if classicComp not in components.components_classic_style:
-            components.components_classic_style.append(classicComp)
         # Create a temporary directory of the source directory to allow
         # tests to modify the source files
-        # TODO: Add tests that modify source files to ensure that errors work
```

Comment:
Could you add this TODO back, please? I think it would be useful to have more fine-grain tests. For example, since this test is using `shutil.copytree` to copy the test folder to tmp, we can modify the `new_style` folder to break in various ways:
- Have it use `impl_suffix` instead of `impl-suffix`
- Call the file `def/interface.go` instead of `def/component.go`
- Use a package name for the impl folder that doesn't match `package <component>impl`.
- etc
Basically this could be a list of checks in the function `check_component_contents_and_file_hiearchy` and we could use this technique to get very high coverage of all of the rules we want to enforce. In other words, I don't expect this TODO to be resolved in this PR, as it's only introducing the single rule of "don't use the classic style" going forward, but there's much more the linter can do.

Reply:
Adding it back in with my own commit

Reply:
Thank you so much for the explanation. 
My reasoning to remove the TODO was because I added a test that check that `get_components_and_bundles` return errors if the component is not path of the exception. But you are absolutely right, that we need to validate issues with components that are not part of the exception lists 😄